src/NServiceBus.RabbitMQ/Delays/DelayInfrastructure.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace NServiceBus.Transport.RabbitMQ
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Text;
+    using global::RabbitMQ.Client;
+
+    static class DelayInfrastructure
+    {
+        const int maxNumberOfBitsToUse = 28;
+        const int maxLevel = maxNumberOfBitsToUse - 1;
+        public const int MaxDelayInSeconds = (1 << maxNumberOfBitsToUse) - 1;
+        public const string DelayHeader = "NServiceBus.Transport.RabbitMQ.DelayInSeconds";
+        public const string DeadLetteredMessageHeader = "x-death";
+        public const string DeliveryExchange = "nsb.delay-delivery";
+
+        public static string LevelName(int level) => $"nsb.delay-level-{level:D2}";
+
+        public static string BindingKey(string address) => $"#.{address}";
+
+        public static void Build(IModel channel)
+        {
+            var bindingKey = "1.#";
+
+            for (var i = maxLevel; i >= 0; i--)
```

Comment:
`for` loops are the one place where I tend to prefer sticking with `i`, `j`, etc for variable names, but that's more out of tradition than anything. Seems like `level` could work here (and the other loops) well enough if you'd prefer to have them changed.